Mathlib/Algebra/Homology/ShortComplex/Preadditive.lean

```diff
@@ -558,6 +558,168 @@ def ofNullHomotopic (h₀ : S₁.X₁ ⟶ S₂.X₁) (h₀_f : h₀ ≫ S₂.f =
+    (h₁ : S₁.X₂ ⟶ S₂.X₁) (h₂ : S₁.X₃ ⟶ S₂.X₂) (h₃ : S₁.X₃ ⟶ S₂.X₃) (g_h₃ : S₁.g ≫ h₃ = 0) :
+    rightHomologyMap' (nullHomotopic _ _ h₀ h₀_f h₁ h₂ h₃ g_h₃) H₁ H₂ = 0 :=
+  (RightHomologyMapData.ofNullHomotopic H₁ H₂ h₀ h₀_f h₁ h₂ h₃ g_h₃).rightHomologyMap'_eq
+
+@[simp]
+lemma homologyMap'_nullHomotopic
+    (H₁ : S₁.HomologyData) (H₂ : S₂.HomologyData)
+    (h₀ : S₁.X₁ ⟶ S₂.X₁) (h₀_f : h₀ ≫ S₂.f = 0)
+    (h₁ : S₁.X₂ ⟶ S₂.X₁) (h₂ : S₁.X₃ ⟶ S₂.X₂) (h₃ : S₁.X₃ ⟶ S₂.X₃) (g_h₃ : S₁.g ≫ h₃ = 0) :
+    homologyMap' (nullHomotopic _ _ h₀ h₀_f h₁ h₂ h₃ g_h₃) H₁ H₂ = 0 :=
+  by apply leftHomologyMap'_nullHomotopic
+
+variable (S₁ S₂)
+
+@[simp]
+lemma leftHomologyMap_nullHomotopic [S₁.HasLeftHomology] [S₂.HasLeftHomology]
+    (h₀ : S₁.X₁ ⟶ S₂.X₁) (h₀_f : h₀ ≫ S₂.f = 0)
+    (h₁ : S₁.X₂ ⟶ S₂.X₁) (h₂ : S₁.X₃ ⟶ S₂.X₂) (h₃ : S₁.X₃ ⟶ S₂.X₃) (g_h₃ : S₁.g ≫ h₃ = 0) :
+    leftHomologyMap (nullHomotopic _ _ h₀ h₀_f h₁ h₂ h₃ g_h₃) = 0 :=
+  by apply leftHomologyMap'_nullHomotopic
+
+@[simp]
+lemma rightHomologyMap_nullHomotopic [S₁.HasRightHomology] [S₂.HasRightHomology]
+    (h₀ : S₁.X₁ ⟶ S₂.X₁) (h₀_f : h₀ ≫ S₂.f = 0)
+    (h₁ : S₁.X₂ ⟶ S₂.X₁) (h₂ : S₁.X₃ ⟶ S₂.X₂) (h₃ : S₁.X₃ ⟶ S₂.X₃) (g_h₃ : S₁.g ≫ h₃ = 0) :
+    rightHomologyMap (nullHomotopic _ _ h₀ h₀_f h₁ h₂ h₃ g_h₃) = 0 :=
+  by apply rightHomologyMap'_nullHomotopic
+
+@[simp]
+lemma homologyMap_nullHomotopic [S₁.HasHomology] [S₂.HasHomology]
+    (h₀ : S₁.X₁ ⟶ S₂.X₁) (h₀_f : h₀ ≫ S₂.f = 0)
+    (h₁ : S₁.X₂ ⟶ S₂.X₁) (h₂ : S₁.X₃ ⟶ S₂.X₂) (h₃ : S₁.X₃ ⟶ S₂.X₃) (g_h₃ : S₁.g ≫ h₃ = 0) :
+    homologyMap (nullHomotopic _ _ h₀ h₀_f h₁ h₂ h₃ g_h₃) = 0 :=
+  by apply homologyMap'_nullHomotopic
+
+namespace Homotopy
+
+variable {φ₁ φ₂ S₁ S₂}
+
+lemma congr_leftHomologyMap' (h : Homotopy φ₁ φ₂) (h₁ : S₁.LeftHomologyData)
+    (h₂ : S₂.LeftHomologyData) : leftHomologyMap' φ₁ h₁ h₂ = leftHomologyMap' φ₂ h₁ h₂ := by
+  rw [h.eq_add_nullHomotopic, leftHomologyMap'_add, leftHomologyMap'_nullHomotopic, add_zero]
+
+lemma congr_rightHomologyMap' (h : Homotopy φ₁ φ₂) (h₁ : S₁.RightHomologyData)
+    (h₂ : S₂.RightHomologyData) : rightHomologyMap' φ₁ h₁ h₂ = rightHomologyMap' φ₂ h₁ h₂ := by
+  rw [h.eq_add_nullHomotopic, rightHomologyMap'_add, rightHomologyMap'_nullHomotopic, add_zero]
+
+lemma congr_homologyMap' (h : Homotopy φ₁ φ₂) (h₁ : S₁.HomologyData)
+    (h₂ : S₂.HomologyData) : homologyMap' φ₁ h₁ h₂ = homologyMap' φ₂ h₁ h₂ := by
+  rw [h.eq_add_nullHomotopic, homologyMap'_add, homologyMap'_nullHomotopic, add_zero]
+
+lemma congr_leftHomologyMap (h : Homotopy φ₁ φ₂) [S₁.HasLeftHomology] [S₂.HasLeftHomology] :
+    leftHomologyMap φ₁ = leftHomologyMap φ₂ :=
+  h.congr_leftHomologyMap' _ _
+
+lemma congr_rightHomologyMap (h : Homotopy φ₁ φ₂) [S₁.HasRightHomology] [S₂.HasRightHomology] :
+    rightHomologyMap φ₁ = rightHomologyMap φ₂ :=
+  h.congr_rightHomologyMap' _ _
+
+lemma congr_homologyMap (h : Homotopy φ₁ φ₂) [S₁.HasHomology] [S₂.HasHomology] :
+    homologyMap φ₁ = homologyMap φ₂ :=
+  h.congr_homologyMap' _ _
```

Comment:
I think I would put the `congr` at the end of the name, instead of at the start. I feel like that way there also a bigger chance of discovering it via autocompletion when searching for names about `leftHomologyMap` etc...
What do you think?

Reply:
Thanks for the suggestion. I have made `congr` a suffix here.